src/main/java/com/openlattice/shuttle/dates/TimeZones.java

```diff
@@ -30,8 +39,34 @@
     public static final TimeZone America_LosAngeles = TimeZone.getTimeZone( "America/Los_Angeles" );
     public static final TimeZone America_Chicago    = TimeZone.getTimeZone( "America/Chicago" );
     public static final TimeZone America_Denver     = TimeZone.getTimeZone( "America/Denver" );
+    public static final TimeZone UTC                = TimeZone.getTimeZone( "UTC" );
 
     private TimeZones() {
     }
 
+    private static final Logger logger = LoggerFactory.getLogger( TimeZones.class );
+
+    public static Optional<TimeZone> checkTimezone( Optional<String> timezoneString ) {
+        if ( timezoneString.isPresent() ) {
+            String timezoneId = timezoneString.get();
+
+            Optional<TimeZone> timezone = Optional.of(TimeZone.getTimeZone( timezoneId ));
+
+            if ( timezone.orElse( Constants.DEFAULT_TIMEZONE ).getID().equals( timezoneId ) ) {
+                throw new IllegalArgumentException(
+                        "Invalid timezone id " + timezoneId + " requested" );
+            }
+            return timezone;
+
+        }
+
+        return Optional.empty();
+    }
+
+    public static void checkTimezonesMatch( OffsetDateTime odt, ZoneId tzId ) {
+        if ( tzId != odt.getOffset() ) {
+            logger.error( "The reported (" + tzId + ") and requested (" + odt.getOffset() + ") timezones are inconsistent." );
```

Comment:
Extremely minor optimization, but you can log like this:
logger.error( "The reported ({}) and requested ({}) timezones are inconsistent.", tzId, odt.getOffset() );

Reply:
Where the curly braces will be replaces with the additional function arguments in order